core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -94,6 +105,188 @@ static CloseableIterable<FileScanTask> planFiles(
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult < 0);
+      }
+
+      @Override
+      public <T> Boolean ltEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult <= 0);
+      }
+
+      @Override
+      public <T> Boolean gt(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult > 0);
+      }
+
+      @Override
+      public <T> Boolean gtEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult >= 0);
+      }
+
+      @Override
+      public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult == 0);
+      }
+
+      @Override
+      public <T> Boolean notEq(BoundReference<T> ref, Literal<T> lit) {
+        return compareDateFileContent(ref, lit, compareResult -> compareResult != 0);
+      }
+
+      @Override
+      public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
+        if (isDateFileContent(ref)) {
+          if (!literalSet.contains(manifestContentId)) {
+            return ROWS_CANNOT_MATCH;
+          }
+        }
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean notIn(BoundReference<T> ref, Set<T> literalSet) {
+        if (isDateFileContent(ref)) {
+          if (literalSet.contains(manifestContentId)) {
+            return ROWS_CANNOT_MATCH;
+          }
+        }
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      @Override
+      public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      /**
+       * Comparison of data file content and literal, using integer comparator.
+       *
+       * @param ref bound reference, comparison attempted only if reference is for date_file.content
+       * @param lit literal value to compare with date file content.
+       * @param desiredResult function to apply to int comparator result, returns true if result is
+       *     as expected.
+       * @return false if comparator does not achieve desired result, true otherwise
+       */
+      private <T> Boolean compareDateFileContent(
+          BoundReference<T> ref, Literal<T> lit, Function<Integer, Boolean> desiredResult) {
+        if (isDateFileContent(ref)) {
+          Literal<Integer> intLit = lit.to(Types.IntegerType.get());
+          int literalValueToCompare = toManifestContentId(intLit.value());
+          int cmp = intLit.comparator().compare(manifestContentId, literalValueToCompare);
```

Comment:
I think this is a bit fragile.  We are just depending on the fact that both Manifest content and dataFile content , id of data is less than id of delete.  I would suggest not to support gt/lt pushdowns for that reason (you can always express it in terms of eq/neq)

Can we not use comparator, and just use more hard coded equality cases?  

ie,
```
if (fileContentLit == DATA)
   return manifestContent == DATA
else 
   return manifestContent == DELETE
```